src/components/organisms/AssetActions/Compute/FormComputeDataset.tsx

```diff
@@ -128,6 +130,13 @@ export default function FormStartCompute({
       if (consumable) setAlgorithmConsumableStatus(consumable.status)
     }
     checkIsConsumable()
+
+    const { result, status } = ocean.assets.checkCredential(
+      algorithmDDO,
+      CredentialType.address,
+      accountId
+    )
+    if (status === 2 || status === 3) setIsWhitelisted(result)
```

Comment:
issue/question(blocking): can this be moved to the `ButtonBuy` component? We already pass `algorithmConsumableStatus`.